kubernetes/package-kafka/kafkaprovider.yml

```diff
@@ -0,0 +1,56 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kafkaprovider
+  namespace: openwhisk
+  labels:
+    name: kafkaprovider
+spec:
+  selector:
+    name: kafkaprovider
+  ports:
+    - port: 5000
+      targetPort: 8091
+      name: kafkaprovider
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: kafkaprovider
+  namespace: openwhisk
+  labels:
+    name: kafkaprovider
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: kafkaprovider
+    spec:
+      restartPolicy: Always
+
+      containers:
+      - name: kafkaprovider
+        imagePullPolicy: IfNotPresent
+        # custom alarm provider image. Not sure what we want to do with this.
+        image: openwhisk/kafkaprovider
+        ports:
+        - name: kafkaprovider
+          containerPort: 8080
+        env:
+        - name: "DB_URL"
+          value: "http://couchdb.openwhisk:5984"
```

Comment:
avoid needing DNS: use "$(COUCHDB_SERVICE_HOST):$(COUCHDB_SERVICE_PORT_COUCHDB)"

Reply:
I updated as you suggested. But I don't quite understand when $(COUCHDB_SERVICE_HOST) was set to the correct value ? I didn't find such key in configmap.

Reply:
kube defines environment variables like these for the hosts/ports of each service.  They are automatically defined in every pod that is created after the service is defined.